### DISCUSSION TITLE
Console search (cmd+f)

### DISCUSSION THREAD

**Discussion by jasonLaster** [RecordReplay/devtools#3803] [Ideas]
The console should have a simple cmd+f search ui

**Comment by jonbell-lot23** [comment]
I'm supportive of this. Let me think through some things:

* We do have a filter output field across the top
* We default to putting a cursor in the console, so pressing cmd-f would default to console search rather than editor search, which means we'd need to think through focus states

Filtering and searching are two different things, of course. But do we think they're different enough to justify the collisions and overlap with editor?

**Reply by bvaughn** [reply]
@jonbell-lot23: I'm going to look into this ticket. Let me know if you'd like to sync up on this when you sign on today. If you're busy that's okay too. I'm happy to just take a pass at it.

**Reply by jasonLaster** [reply]
With tickets like this, it's good to start with chrome and firefox devtools parity. Not to say we shouldnt do different things, but that's the baseline

**Reply by markerikson** [reply]
Huh. Y'know, I've never even _tried_ to do a "Find" in a browser console.

Pulling out Chrome right now, my first two observations are:

- There's a "Filter" toolbar up top that's permanently visible and does exactly what ours does right now (filter while you type)
- If you hit CTRL-F, there's a second search bar that opens up at the bottom.  It highlights the matching hits, shows the number of hits, up/down arrows, etc


![image](https://user-images.githubusercontent.com/1128784/163479731-a9482f96-9403-49a8-a1c1-1bdb593d62e7.png)

**Reply by bvaughn** [reply]
> With tickets like this, it's good to start with chrome and firefox devtools parity. Not to say we shouldnt do different things, but that's the baseline

Yes. This was my assumption. 😄 

> Huh. Y'know, I've never even tried to do a "Find" in a browser console.

I've mostly used the filter functionality in the past as well, even though I think the search functionality is often what I _actually want_.

**Comment by markerikson** [comment]
Asking for feedback on Twitter:

https://twitter.com/acemarke/status/1514728111769886722

Will inline any responses:

> When browsing Network tab I was surprised to find that ctrl-f started searching sources instead of filtering network calls.

**Reply by bvaughn** [reply]
Okay 😄 

I have a pretty good idea of how this should work FWIW. My message to Jon above was basically just inviting discussion. I'm currently planning on roughly mirroring browser behavior. Should have POC to share by tomorrow.
